### ISSUE TITLE
Date pickers don't support screen readers

### ISSUE THREAD

**Issue by Joseph-Alley** [mantinedev/mantine#2869]
### What package has an issue

@mantine/dates

### Describe the bug

I've been trying to add accessibility to a web app, and although Mantine has been incredible in most places, I've found that it falls short on the date picker inputs. (I'm using the date range picker in particular, but they all seem to have the same issues)

For context, I'm experimenting with VoiceOver on a Mac, using both Chrome and Safari. Here's what seems wrong to me, although I'm not an accessibility expert:

- When highlighting the date picker, the element should have a description to explain that this is a date picker.
- Using Voiceover to activate the date picker should open the date picker 
- - This didn't work on Chrome
- - This did work on Safari, however, so might be a browser issue
- The clear button should have a label to make it clear what it does
- Next and previous month buttons should also have labels
- Month selector should have a more detailed label explaining that it can be clicked to allow choosing another month
- - Ideally, when moving between day and month screens, there should be some alert to explain.
- When navigating the calendar, each date button should say the full date, rather than just a number
- Similarly the month pickers should say the full month.
- In general, there should be more descriptions to explain the state of the component.

There's likely other issues that would come up during the process of working on these.
  
  I found the following examples of a date picker being somewhat more accessible, for comparison.:
- https://github.com/react-dates/react-dates
- https://dequeuniversity.com/library/aria/date-picker

### What version of @mantine/hooks page do you have in package.json?

^5.6.4

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/strange-pond-6idf8l?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
> When highlighting the date picker, the element should have a description to explain that this is a date picker.

You can add `aria-describedby` or `aria-description` attributes to the input, Mantine does not provide any default labels as we want to avoid introducing localized content:

```tsx
<DatePicker aria-describedby="id" aria-description="This is date picker" />
```

Usually, these props are not required as input should be described by the label:

```tsx
<DatePicker label="Pick dates" />
```


> The clear button should have a label to make it clear what it does

Set label on your side with `clearButtonLabel` prop:

```tsx
<DatePicker clearButtonLabel="Clear date" />
```

> Next and previous month buttons should also have labels

See props table, add these labels with `nextYearLabel`, `nextMonthLabel`, `nextDecadeLabel`, `previousYearLabel`, `previousMonthLabel`, `previousDecadeLabel` props.


All other issues will be addressed in 6.0 revamp

**Comment by rtivital** [comment]
Fixed in 6.0
